assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -0,0 +1,258 @@
+					areaWidgets[ areaSlug ] = [];
+				}
+
+				if ( ! areaWidgets[ areaSlug ].includes( slug ) ) {
+					areaWidgets[ areaSlug ].push( slug );
+				}
+			} );
+
+			return {
+				...state,
+				areaWidgets,
+			};
+		}
+
+		case REGISTER_WIDGET: {
+			const { slug, settings } = payload;
+
+			if ( state.widgets[ slug ] !== undefined ) {
+				global.console.warn( `Could not register widget with slug "${ slug }". Widget "${ slug }" is already registered.` );
+
+				return { ...state };
+			}
+
+			return {
+				...state,
+				widgets: {
+					...state.widgets,
+					[ slug ]: { ...settings, slug },
+				},
+			};
+		}
+
+		case SET_WIDGET_COMPONENT_KEY: {
+			const { registryKey } = payload;
+
+			return {
+				...state,
+				registryKey,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+};
+
+export const selectors = {
+	/**
+	 * Checks if a widget has been registered with a given slug.
+	 *
+	 * Returns `true` if the widget area has been registered.
+	 * Returns `false` if the widget area has NOT been registered.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {Object}  state Data store's state.
+	 * @param  {string}  slug  Widget's slug.
+	 * @return {boolean}       `true`/`false` based on whether widget has been registered.
+	 */
+	isWidgetRegistered( state, slug ) {
+		const { widgets } = state;
+
+		return widgets[ slug ] !== undefined;
+	},
+
+	/**
+	 * Returns all widgets registered for a given widget area.
+	 *
+	 * Returns an array of all widgets for a given area.
+	 * The widgets are returned in order of their priority, so can be rendered in
+	 * the order provided by the selector.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {Object} state          Data store's state.
+	 * @param  {string} widgetAreaSlug Widget context to get areas for.
+	 * @return {Array}                 An ordered array of widgets for this area.
+	 */
+	getWidgets: createRegistrySelector( ( select ) => ( state, widgetAreaSlug ) => {
+		invariant( widgetAreaSlug, 'widgetAreaSlug is required.' );
+
+		const { areaWidgets, widgets } = state;
+
+		const registryKey = select( STORE_NAME ).getWidgetRegistryKey();
+
+		return Object.values( widgets ).filter( ( widget ) => {
+			return areaWidgets[ widgetAreaSlug ] && areaWidgets[ widgetAreaSlug ].includes( widget.slug );
+		} ).sort( ( widgetA, widgetB ) => {
+			return ( widgetA.priority >= widgetB.priority ) ? 1 : -1;
+		} ).map( ( widget ) => {
+			const widgetWithComponent = { ...widget };
+			if ( WidgetComponents[ registryKey ] ) {
+				widgetWithComponent.component = WidgetComponents[ registryKey ][ widget.slug ];
+			}
+
+			return widgetWithComponent;
+		} );
+	} ),
+
+	/**
+	 * Returns the registry key being used for this registry's widgets.
+	 *
+	 * We key each registry with an Integer, so we don't share registered widgets
+	 * between registries. This allows us to access the appropriate registry global
+	 * from inside selectors.
+	 *
+	 * @private
+	 * @since n.e.x.t
```

Comment:
Nitpick, but `private` should follow `since`.

Reply:
Ah, I didn't know we had a preference on that one! 👍